gpu_time/gpu_time_test.cpp

```diff
@@ -0,0 +1,434 @@
+/*
+Copyright 2025 Google Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include <gtest/gtest.h>
+#include "gpu_time.h"
+
+namespace
+{
+
+// Since the GPUTime class relies on Vulkan types and function pointers,
+// we need to create mocks and stubs for them to use in our tests.
+// We are not testing Vulkan itself, just how our class interacts with it.
+
+// --- Mock Vulkan Handles and Functions ---
+
+// Use custom non-zero pointer values for handles to simulate real ones.
+// Using reinterpret_cast to create opaque handles from integer values.
+#define MOCK_HANDLE(name, type, val) \
+    const type name = reinterpret_cast<type>(static_cast<uintptr_t>(val));
+
+MOCK_HANDLE(MOCK_DEVICE, VkDevice, 0x1);
```

Comment:
nit: consider swapping mame and type so it loops a little more like a normal variable declaration

```suggestion
MOCK_HANDLE(VkDevice, MOCK_DEVICE, 0x1); // reads more like VkDevice MOCK_DEVICE = 0x1;
```